### PR TITLE
Change kernel/user stacks order

### LINKED ISSUE #559: offcputime user/kernel stack order
```
# ./offcputime.py -p 21077
Tracing off-CPU time (us) of PID 21077 by user + kernel stack... Hit Ctrl-C to end.
^C
    00007f4ff2a4f810 __read
    000000000040866f read_token
    000000000040816d yylex
    000000000040580e yyparse
    000000000040546b parse_command
    000000000040553c read_command
    0000000000405270 reader_loop
    00000000004038e7 main
    00007f4ff2985ec5 __libc_start_main
    ffffffff810a16d1 finish_task_switch
    ffffffff817b3255 schedule
    ffffffff817b5d1b schedule_timeout
    ffffffff810bc0f3 wait_woken
    ffffffff814b6308 n_tty_read
    ffffffff814b0df2 tty_read
    ffffffff81205168 __vfs_read
    ffffffff81205fcf vfs_read
    ffffffff812073c6 sys_read
    ffffffff817b6d76 entry_SYSCALL_64_fastpath
    -                bash (21077)
        124267
```

Note that the output begins with user, then shows kernel. I think it should be the other way around, kernel first, then user, to preserve the logical code flow from the bottom of the stack to the top. You always begin with "main" at the bottom, then work up, and sometimes enter the kernel.

Same with folded:

```
# ./offcputime.py -fp 21077
^Cbash;entry_SYSCALL_64_fastpath;sys_read;vfs_read;__vfs_read;tty_read;n_tty_read;wait_woken;schedule_timeout;schedule;__libc_start_main;main;reader_loop;read_command;parse_command;yyparse;yylex;read_token 239532
```

Oops, another bug (maybe mine): note that the folded output is missing "__read" and "finish_task_switch" -- must be an off-by-one bug in the loops.

I'd also add an option to insert a delimiter frame, "-", which is supported by flamegraph.pl (it colors it grey). Eg, as -d:

```
# ./offcputime.py -d -p 21077
Tracing off-CPU time (us) of PID 21077 by user + kernel stack... Hit Ctrl-C to end.
^C
    ffffffff810a16d1 finish_task_switch
    ffffffff817b3255 schedule
    ffffffff817b5d1b schedule_timeout
    ffffffff810bc0f3 wait_woken
    ffffffff814b6308 n_tty_read
    ffffffff814b0df2 tty_read
    ffffffff81205168 __vfs_read
    ffffffff81205fcf vfs_read
    ffffffff812073c6 sys_read
    ffffffff817b6d76 entry_SYSCALL_64_fastpath
    - -
    00007f4ff2a4f810 __read
    000000000040866f read_token
    000000000040816d yylex
    000000000040580e yyparse
    000000000040546b parse_command
    000000000040553c read_command
    0000000000405270 reader_loop
    00000000004038e7 main
    00007f4ff2985ec5 __libc_start_main
    -                bash (21077)
        124267

# ./offcputime.py -d -fp 21077
^Cbash;__libc_start_main;main;reader_loop;read_command;parse_command;yyparse;yylex;read_token;-;entry_SYSCALL_64_fastpath;sys_read;vfs_read;__vfs_read;tty_read;n_tty_read;wait_woken;schedule_timeout;schedule 239532
```

/CC @evverx

### PR DESCRIPTION
**Pull request by evverx**
* offcputime: change user/kernel stack order
* offcputime: add the `-d` option

Closes #559

### CODE REVIEW

File: tools/offcputime.py [hunk 1]
```diff
@@ -61,6 +61,8 @@ def positive_nonzero_int(val):
     help="show stacks from user space only (no kernel space stacks)")
 stack_group.add_argument("-K", "--kernel-stacks-only", action="store_true",
     help="show stacks from kernel space only (no user space stacks)")
+parser.add_argument("-d", "--delimited", action="store_true",
```
Comment (abirchall) [inline tools/offcputime.py]: Per my comment in #559 and the plans regarding #545 we may want to be more verbose with the long version of this option. Possibly `--delimit-user-kernel-stacks` but I'm terrible at naming so there may be a better alternative haha. What do you think?

**Comment by abirchall** [comment]
LGTM

nit: We should update the comment with the arg format at the top of `tools/offcputime.py`, in the examples and in the man file.

File: tools/offcputime.py (hunk 1 above)
Comment (evverx) [inline tools/offcputime.py]: @abirchall , thanks for the review!
oh, I didn't see #545 . I'll take a look

**Comment by brendangregg** [comment]
@abirchall thanks, yes, I've made a note (and another tool needs USAGE args updated too).

I think the default output delimiter should be "- -" not "--", to aid people parsing it and expecting two columns, but that's another minor nit. The folded delimiter should be "-" and is.

**Pull request merged**